pyvista/utilities/reader.py

```diff
@@ -988,6 +988,40 @@ def all_patch_arrays_status(self):
         return {name: self.patch_array_status(name) for name in self.patch_array_names}
 
 
+class POpenFOAMReader(OpenFOAMReader):
+    """POpenFOAM Reader for .foam files."""
+
+    _class_reader = staticmethod(_vtk.lazy_vtkPOpenFOAMReader)
+
+    @property
+    def case_type(self):
+        """Indicate whether decomposed mesh or reconstructed mesh should be read.
+
+        Returns
+        -------
+        int
+            If ``1``, reconstructed mesh should be read. If ``0``, decomposed mesh should be read.
+
+        Examples
+        --------
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> filename = examples.download_cavity(load=False)
+        >>> reader = pyvista.POpenFOAMReader(filename)
+        >>> reader.case_type = 1
+        >>> reader.case_type
+        1
+        """
+        return self.reader.GetCaseType()
+
+    @case_type.setter
+    def case_type(self, value):
+        if value:
+            self.reader.SetCaseType(1)
+        else:
+            self.reader.SetCaseType(0)
+
```

Comment:
In my testing we need to call `self._update_information` after setting `case_type`.
```suggestion
        self._update_information()

```

Reply:
Ok, added!